### DISCUSSION TITLE
Q1 Error factor of 10

### DISCUSSION THREAD

**Discussion by propene123** [wenceorg/comp4187#94] [Q&A] [answered]
Hi, for question 1 I have discretised the spatial operator as in the first coursework as suggested here https://github.com/wenceorg/comp4187/discussions/80#discussioncomment-2432620. But for some reason when comparing my numerical solutions to the exact solution there are off by a factor of 10 every time. I can solve this by scaling the matrix for the spatial operator, but I don't see any mathematical reason as to why this scaling is necessary (especially since the rhs is not scaled with it). Is it possible that the analytical solution given in the question is wrong by a factor of 10 somewhere?

**Comment by Julian-Wyatt** [comment]
I've got a similar problem - ie I need to have `Hx = 1 / (mx - 2)` in addition to multiplying matrix by Hx, and leaving the rhs scaled as 1
Which gives near perfect results for refinements 1,...,9, but breaks down after that - while this doesn't work at all when solving with the solver from Q3a - quickly diverges

To clarify, this is also for the Euler Bernoulli beam

**Reply by propene123** [reply]
Yep this is exactly the issue I have, with the same scaling

**Reply by shadowbourne** [reply]
> Yep this is exactly the issue I have, with the same scaling

@propene123 do you also observe better convergence with Hx = 1/(mx-2) vs the usual Hx = 1/(mx-1) ?

This whole question makes no sense theory wise to me, everything that seems wrong works but the "correct" things break 😭

**Reply by annereinarz** [reply]
Since you need to set Hx like that. Is it perhaps an issue with the counting/not counting the Dirichlet conditions? E.g. have you removed those from the matrix or are they included and set to zero?

**Reply by propene123** [reply]
I personally haven't had to set Hx like that, the normal way works for me. But I have had to multiply by Hx which makes no sense to me as I cant find a mathematical reason for it since multigrid isn't be used

**Reply by annereinarz** [reply]
Just to make sure I'm understanding the issue correctly. If you multiply the matrix with Hx, i.e. if you use a scaling factor of Hx^3 you get the correct solution? Or there is a factor of 10 missing as well?

**Reply by propene123** [reply]
So my solutions seemed to be out by factor of 10, but for some reason multiplying the matrix by an additional Hx fixes this. However I don't see any reason why this is necessary.

**Reply by annereinarz** [reply]
It shouldn't be. In my tests the analytical solution appears correct.

Just to check:
- You are using the same matrix that you used in coursework 1 (just moved to petsc)?
- The point source can be a little tricky, e.g. if you have an even number of points you'd need to make sure that the total force still sums to 1. And even for an odd number of points (which you seem to be using) you're actually approximating the point source by a function that is not a point source. The important thing is that the total force is 1. This means the force at the central point should be  1/Hx. Is this perhaps where you've lost a factor?
- For the multigrid convergence note that the solution to this problem is not smooth. This will affect the behavior of the solver and you can't expect "textbook multigrid convergence".

**Reply by propene123** [reply]
Ah yes after thinking about that, it makes sense. It's that the point source's contribution to the point next to it is spread out over an "area" of h_x so must be scaled.

**Comment by annereinarz** [comment, marked as answer]
I'm marking this resolved.

The issue seems to be a missing 1/Hx factor in the right hand side.
